misk-aws/src/main/kotlin/misk/jobqueue/sqs/AwsSqsJobQueueConfig.kt

```diff
@@ -52,5 +54,61 @@ class AwsSqsJobQueueConfig(
    * We use the default retry strategy with SQS, which retries 3 times.
    * As a result, your app could potentially spend 3 x this timeout talking to SQS.
    */
-  val sqs_sending_request_timeout_ms: Int = 5000
+  val sqs_sending_request_timeout_ms: Int = 5000,
+
+  val aws_sqs_job_receiver_policy: AwsSqsJobReceiverPolicy = AwsSqsJobReceiverPolicy.ONE_FLAG_ONLY
 ) : Config
+
+
+/**
+ * AWS SQS consumers are spun by each of a service's pods. Each pod is responsible for running some
+ * number on consumers and making sure they are within the parameters of the feature-flags used.
+ *
+ * Which flags?
+ *  * [CONSUMERS_PER_QUEUE]: This flag specifies the total number of consumers across the entire
+ *    cluster.
+ *  * [POD_CONSUMERS_PER_QUEUE]: This flag specifies the number of consumers a single pod should
+ *    have.
+ *
+ * The [AwsSqsJobReceiverPolicy] gives two options for how consumers are created based on the flags.
+ */
+enum class AwsSqsJobReceiverPolicy {
+  /**
+   * This is the original policy. Naming is hard, but this policy will compute receivers as follows.
+   * First we choose one flag. If there is a configuration in [POD_CONSUMERS_PER_QUEUE], choose that
+   * flag; otherwise choose the [CONSUMERS_PER_QUEUE] flag.
+   *
+   * If the [POD_CONSUMERS_PER_QUEUE] is chosen, ALL pods will spin up the configured number of
+   * consumers. Imagine the flag is configured for 5 consumers, then
+   *  5 pods => 25 sqs consumers
```

Comment:
is there a reason why we cannot impose a "default" max per pod and then if a service wants to increase it, they do it as an explicit action ? I know that the way to arrive at that number though.

Reply:
yeah this is a good point

I think the part I want to get right is how we make sure we don't break existing things by keeping things backwards compatible and so I didn't add any defaults to the existing one.

But good catch that we should add a default given that folks might use the BALANCED_MAX policy even if the feature flag is not set. Like you said the tricky thing is finding that number, but it would reduce the risk + burden of using this policy. (my first choice for defaults is 1 or 5, ie smallest positive number or something that feels reasonable for most deployments based on "anecdata")


We could choose to avoid this config altogether and add a default and just replace the receiver algo with the new policy. I would look to the @misk-maintainers for guidance here since changing the current behavior might be risky.

Reply:
- [ ] handle the case where the per pod flag is configured with a non-positive number, this would make the BALANCED_MAX policy easier to use, especially since that flags default is <=0 which means that sqs wouldn't run at all)